### DISCUSSION TITLE
v0.10.0-beta

### DISCUSSION THREAD

**Discussion by mgdigital** [bitmagnet-io/bitmagnet#316] [Announcements]
This beta release includes a full rewrite of the web UI, incorporating the following features and improvements:

- Translations
- Themes
- Responsive mobile-friendly UI
- Permalinks, closing https://github.com/bitmagnet-io/bitmagnet/issues/192
- Health check within web UI
- Monitoring dashboard
- Queue management within UI
- Allow bulk-copying of magnet links and info hashes
- Separation of torrent files query from main search query, and pagination through files in a torrent
- Show sizes in MiB instead of MB
- Always show original torrent name in table, closing https://github.com/bitmagnet-io/bitmagnet/issues/290
- Add documentation for starting Bitmagnet, closing https://github.com/bitmagnet-io/bitmagnet/issues/312

Please report any issues found. In particular there may be issues with the translations, which were done by AI for the most part! If you are proficient in any of these languages then your help is appreciated in reviewing the translation files here: https://github.com/bitmagnet-io/bitmagnet/tree/main/webui/src/app/i18n/translations

## What's Changed
* add queue jobs metric in grafana by @scratchmex in https://github.com/bitmagnet-io/bitmagnet/pull/297
* Webui revamp by @mgdigital in https://github.com/bitmagnet-io/bitmagnet/pull/280

## New Contributors
* @scratchmex made their first contribution in https://github.com/bitmagnet-io/bitmagnet/pull/297

**Full Changelog**: https://github.com/bitmagnet-io/bitmagnet/compare/v0.9.5...v0.10.0-beta.1

<hr /><em>This discussion was created from the release <a href='https://github.com/bitmagnet-io/bitmagnet/releases/tag/v0.10.0-beta.1'>v0.10.0-beta.1</a>.</em>

**Comment by alexmicuplusfour** [comment]
Congrats on the release!
I'm looking forward to getting it working properly on my machine someday.

Any idea why I get this?

> 2024-10-15 02:25:34 INFO        worker/worker.go:201    started worker  {"key": "http_server"}
> 2024-10-15 02:25:35 INFO        worker/worker.go:201    started worker  {"key": "queue_server"}
> 2024-10-15 02:25:35 INFO        worker/worker.go:201    started worker  {"key": "dht_crawler"}
> 2024-10-15 02:25:35 WARN        dht_crawler     dhtcrawler/bootstrap.go:20      failed to resolve bootstrap node address: lookup dht.anacrolix.link on 127.0.0.11:53: no such host
> 2024-10-15 02:25:35 WARN        dht_crawler     dhtcrawler/bootstrap.go:20      failed to resolve bootstrap node address: lookup router.bittorrent.cloud on 127.0.0.11:53: no such host

**Reply by mgdigital** [reply]
Yes, the bootstrap nodes come from the Anacrolix DHT library and need to be pruned as some are no longer operational - doesn't indicate any problem.

**Reply by alexmicuplusfour** [reply]
ok, thank you! I assumed those messages were an indication of why DHT is down. I've updated to beta 3 and now the warnings are gone, but DHT is still down.

**Reply by mgdigital** [reply]
When you say it's down, do you mean the new health check is showing DHT as down? If so you definitely have a networking issue - check VPN is configured correctly maybe...

**Reply by alexmicuplusfour** [reply]
Yes, the new health check. I don't use a VPN. I'm using Docker, by the way, on windows. No other containers have networking issues. The "# BitTorrent ports" are correctly forwarded on the router.

**Reply by alexmicuplusfour** [reply]
I just discovered that it works fine for a few minutes after windows is restarted. And then it won't work anymore no matter what I do.

**Reply by mgdigital** [reply]
Okay. Would be nice to get to the bottom of it but I don't have a Windows machine to test on. I expect there'd be more bug reports if lots of people were seeing this. Any other Windows users having the same issue?

You might have more luck on WSL than plain Windows if you aren't already...

**Reply by alexmicuplusfour** [reply]
ok, no worries. Thanks!

**Reply by alexmicuplusfour** [reply]
I'm using WSL 2.
Here's another clue. With the default settings the DHT stays on for less than a minute. And if i adjust DHT_CRAWLER_SCALING_FACTOR, the lower it's set, the longer it stays on. At "1", it stays on for a few hours. And then it breaks the networking for all the other containers (they can't access the internet anymore). And to fix it I have to restart docker.

**Reply by alexmicuplusfour** [reply]
I gave up and installed docker on a raspberry pi 🙏

**Reply by MUSESTAR-Official** [reply]
I also have this issue when using version 0.9.5. How to fix it?

**Reply by alexmicuplusfour** [reply]
@MUSESTAR-Official Use linux.

**Reply by mgdigital** [reply]
It does seem some Windows users have issues. It would be good to get to the bottom of these and when I get time I'll fire up a VM to try and reproduce it. For now it's known to work well on Linux and MacOS.

@MUSESTAR-Official , we're discussing the, v.0.10.0 beta here, you might have more luck with that, let me know...

If any Windows users can dig any deeper on these issues it would be appreciated!

**Reply by NiklasBr** [reply]
I noticed that my DHT kept stopping as well on WSL2 in Windows 11 so I tried to change the logging level but I think I have misunderstood some instructions because when I add a `config/config.yaml` file it won't start:

```shell
$ ls -lah
total 32K
drwxr-xr-x  7 niklas niklas 4.0K Dec  5 10:17 .
drwxr-xr-x 26 niklas niklas 4.0K Oct 29 08:34 ..
drwxr-xr-x  8 niklas niklas 4.0K Dec  5 10:06 .git
drwxr-xr-x  4 niklas niklas 4.0K Feb 16  2024 .idea
drwxr-xr-x  2 niklas niklas 4.0K Dec  5 10:17 config
drwxr-xr-x  9 root   root   4.0K Feb 16  2024 data
-rw-r--r--  1 niklas niklas 1.2K Dec  5 10:10 docker-compose.yaml
drwxr-xr-x  9 root   root   4.0K Feb 16  2024 observability

$ ls -lah config/
total 12K
drwxr-xr-x 2 niklas niklas 4.0K Dec  5 10:17 .
drwxr-xr-x 7 niklas niklas 4.0K Dec  5 10:17 ..
-rw-r--r-- 1 niklas niklas   20 Dec  5 10:08 config.yaml

$ cat config/config.yaml
log:
  level: debug
```

Starting:
```
$ docker compose up -d  bitmagnet
[+] Running 1/2
 ✔ Container bitmagnet-postgres  Healthy                                                                                                                                                                                                                                           0.5s
 ⠴ Container bitmagnet           Starting                                                                                                                                                                                                                                          0.9s
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu-20.04/b496ea713a8b8649ee4e306616c0ad6672721d4c52fdb887c82a767bc4e6d6d8" to rootfs at "/root/.config/bitmagnet": mount /run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu-20.04/b496ea713a8b8649ee4e306616c0ad6672721d4c52fdb887c82a767bc4e6d6d8:/root/.config/bitmagnet (via /proc/self/fd/6), flags: 0x5000: no such file or directory: unknown
```

**Reply by tyduptyler13** [reply]
It seems that WSL docker has DNS issues. I suspect that additional config is required to resolve the CNAME type records and recurse through them. The host environment has issues resolving that same host but my ubuntu in WSL has no problem (probably because dig is implemented differently than windows Resolve-DnsName (which is what docker desktop ultimately uses for its dns))

**Reply by mcmikemn** [reply]
I'n getting these exact warnings, too. And though they're only warnings, I can't get bitmagnet to work (I get "bad gateway"). My docker compose stack includes Wireguard and this is all behind Traefik, run on a Debian host. This is all at the very edge of my skillset so likely this isn't a bug but I'm doing something wrong.

I can `dig google.com` from the bitmagnet container and it resolves, so I think I set up the docker networks correctly. `dig dht.anacrolix.link` from the bitmagnet container returns:
```
; <<>> DiG 9.18.33 <<>> dht.anacrolix.link
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 24233
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;dht.anacrolix.link.            IN      A

;; ANSWER SECTION:
dht.anacrolix.link.     300     IN      CNAME   router.bittorrent.cloud.
router.bittorrent.cloud. 300    IN      CNAME   a3600c27-f507-4a67-9c83-929cea1e43c4.pub.instances.scw.cloud.

;; AUTHORITY SECTION:
pub.instances.scw.cloud. 60     IN      SOA     nsa.scaleway.com. hostmaster.pub.instances.scw.cloud. 1739890357 300 3600 604800 60

;; Query time: 275 msec
;; SERVER: 10.64.0.1#53(10.64.0.1) (UDP)
;; WHEN: Tue Feb 18 14:52:58 UTC 2025
;; MSG SIZE  rcvd: 216
````
I see there is no A record - I'm not sure what that means. I get the same results from my local computer so I don't think there's any issue with the VPN (but again, I could be wrong).

I'm not sure if these warnings are related to to bitmagnet returning "bad gateway", but I the only other thing in the logs that might be telling is from postgres:
```
ERROR:  relation "goose_db_version" does not exist at character 36
STATEMENT:  SELECT version_id, is_applied from goose_db_version ORDER BY id DESC
```
Could that be related?

I'd appreciate any help anyone can offer. Let me know if it would be better to move this discussion elsewhere.

**Comment by NiklasBr** [comment]
Nice work on the new UI!

> Show sizes in MiB instead of MB

This is a pet peeve of mine :) 

Rather than defaulting to binary/base 2, is it possible to use the host operating system setting? For example if I am running this on [Ubuntu](https://wiki.ubuntu.com/UnitsPolicy) or [Mac OS](https://support.apple.com/en-us/102119) then those operating systems use decimal for storage size and this will be a bit of a discrepancy.

**Reply by mgdigital** [reply]
So my reasoning here is that pretty much all the other trackers seem to use MiB. I think it might be confusing to change it based on the OS, but it could certainly be added as a user preference (probably this would be for a future release). I'll add a tooltip on mouse-over to show the decimal size in the meantime. Anyone else have thoughts on this?

**Reply by mgdigital** [reply]
Tooltip added in https://github.com/bitmagnet-io/bitmagnet/pull/325
